src/io.ts

```diff
@@ -109,7 +110,17 @@ export function readTodoStorageFile(todoStorageFilePath: string): Operation[] {
  * @param operations - An array of string operations that are used to recreate todos.
  */
 export function writeTodoStorageFile(todoStorageFilePath: string, operations: Operation[]): void {
-  writeFileSync(todoStorageFilePath, operations.join(EOL));
+  writeFileSync(todoStorageFilePath, operations.join(EOL) + EOL);
```

Comment:
When `readTodoStorageFile` called this function, it provided operations with an empty string
`['op1', 'op2', '']`
thats why just `.join(EOL)`worked.

When `compactTodoStorageFile` called this function, it provided an actual array of operations as its supposed to
`['op1', 'op2']`
thats why `.join(EOL)` didn't end with an EOL for compacts.

Now that both functions only call this with an actual array of operations, we can fix the EOL problem here and ensure the final EOL.